src/raftstore/store/peer.rs

```diff
@@ -1255,15 +1266,27 @@ impl Peer {
 
         match change_type {
             ConfChangeType::AddNode => {
-                status.progress.insert(peer.get_id(), Progress::default());
+                if let Some(mut progress) = status.learner_progress.remove(&peer.get_id()) {
+                    // For promote learner to voter.
+                    progress.is_learner = false;
+                    status.progress.insert(peer.get_id(), progress);
+                } else {
+                    status.progress.insert(peer.get_id(), Progress::default());
+                }
             }
             ConfChangeType::RemoveNode => {
+                if peer.get_is_learner() {
+                    // If the node is a learner, we can return directly.
```

Comment:
Do we need to clean the `learner_progress`?

Reply:
The `learner_progress` here is seperated from `RawNode` because `RawNode::status` has already copyed the field. So we don't do that here.